File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>


int main(void)
{
    int letters = 0;
    int words = 0;
    int sentences = 0;

    string text = get_string("Text to be analysed: \n");

    int n = strlen(text);

    // for count of letters
    for (int i = 0; i <= n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    // for words
    for (int i = 0; i <= n; i++)
    {
        if ((isspace(text[i])))
        {
            words++;
        }
    }
    words++;     // as condition is space after word which is not the case for the last word

    // for sentences
    for (int i = 0; i <= n; i++)                    // for accessing each element of an element of the array
    {
        if ((text[i] == 46) || (text[i] == 33) || (text[i] == 63))   // space - 32, ! - 33, ? - 63, . - 46
        {
            sentences++;
        }
    }

    float L = ((letters * 100.00) / words);
    float S = ((sentences * 100.00) / words);

    float answer = 0.0588 * L - 0.296 * S - 15.8;

    answer = round(answer);
    int final = answer;

    if (answer < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (answer > 16 || answer == 16)
    {
        printf("Grade 16+\n");
    }

    else if (answer >= 1 || answer < 16)
    {
        printf("Grade %i\n", final);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

letters = 0
words = 1
sentences = 0

text = get_string("Text to be analysed: ")
n = len(text)
for i in range(n):
    if text[i].isalpha():
        letters += 1
    elif text[i].isspace():
        words += 1
    elif text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences += 1

L = ((letters * 100.00) / words)
S = ((sentences * 100.00) / words)

answer = 0.0588 * L - 0.296 * S - 15.8

answer = round(answer)
final = answer

if answer < 1:
    print("Before Grade 1")
elif answer >= 16:
    print("Grade 16+")
elif answer >= 1:
    print("Grade", answer)
